### ISSUE TITLE
Работа с презентациями

### ISSUE THREAD

**Issue by pafik13** [pafik13/doc-merch#5]
### Функционал для работы с презентациями

Презентации, в данном случае, это материал, который представители будут показывать врачам на визите. Необходим редактор презентаций для менеджеров. Приблизительный список функционала следующий:
- добавление/редактирование презентаций
- версионирование презентаций
- просмотр презентаций
- шаблоны
- хранение картинок

**Comment by pafik13** [comment]
Сейчас прикидываю структуру данных и дизайн. Скоро поделюсь :) Презентации будут состоять во многом из картинок, поэтому 100% понадобиться сервис для работы с ними. В связи с этом можешь начинать изучать функционал [этого сервиса](http://cloudinary.com/).

**Comment by pafik13** [comment]
Так. У меня не было много времени подумать, но затягивать мне не хочется. Необходимо начать с малого: с одного стандартного шаблона. Состоять он должен только из картинок и иметь две панели навигации - верхняя и нижняя. Верхняя панель - это кнопки, которые ассоциированы с некоторыми тематическими группами, а нижние - это соответствующие подгруппы. В подгруппу входит несколько картинок. Примерный вид презентации и пустой шаблон я попытался изобразить [здесь](http://slides.com/pafik13/deck/fullscreen#/). Я там немного намудрил с появлением пустого шаблона, но это смысла не несет :)

**Comment by pafik13** [comment]
Исходя из выше сказанного и показанного, я вижу следующую структуру в данных:
1. Презентация: название, шаблон, дата создания, создатель
2. Шаблон: название, путь к шаблону
3. Группа: название
4. Подгрупп: название, группа
5. Слайд: картинка, подгруппа, очередь

Шаблон предлагаю  сейчас сделать 1 и выставлять его по умолчанию(в дальнейшем будет диалог выбора). Вставлять его в twig можно при помощи директивы {% include %}.

**Comment by dashadasha172** [comment]
Что-то не уверена, что правильно поняла про шаблон и навигацию) В шаблон загружается картинка из слайда, группы, подгруппы; кнопками навигации загружаются другие слайды из соответствующей группы/подгруппы?

**Comment by dashadasha172** [comment]
в структуре "Презентация" должна быть сслыка на массив слайдов?

**Comment by pafik13** [comment]
1. Навигация: 
   - при выборе группы должен отобразиться первый слайд из первой подгруппы
   - при выборе подгруппы должен отобразиться первый слайд этой подгруппы
   - слайды можно листать, при этом будут переходы из одной подгруппы в другую и из группы в другую группу
2. По желанию) Пока не вижу прямой необходимости в этом, так как в текущей ситуации получается в одной презентации много групп, в одной группе много подгрупп, в одной подгруппе много слайдов. Всё связано и в принципе от презентации до слайдов можно дойти. Такая ссылка нужна будет скорее всего для оптимизации скорости или как фишка при реализации. Следовательно, на Ваше усмотрение.

**Comment by pafik13** [comment]
## ТЗ - Презентации

### Схема редактора

Создание и редактирование презентации должно происходить через визуальный конструктор, то есть изменения должны быть сразу же отображены (по своей сути [WYSIWYG](https://ru.wikipedia.org/wiki/WYSIWYG)). Схема редактора изображена на первом слайде [здесь](http://slides.com/pafik13/deck-1). Здесь представлены:
- содержимое презентации (на схеме под номером 1) - здесь должен быть отображен текущий слайд или "пустой слайд";
- верхняя панель (на схеме под номером 2) - отвечает за логические группы презентации и представляет собой группу кнопок;
- верхняя боковая панель (на схеме под номером 3) - отвечает за управление верхней панелью в режиме редактирования;
- нижняя панель (на схеме под номером 4) - отвечает за логические подгруппы презентации и представляет собой группу кнопок;
- нижняя боковая панель (на схеме под номером 5) - отвечает за управление нижней панелью в режиме редактирования;
- навигационные стрелки (на схеме под номером 6) - отвечают за перелистывание презентации вперед-назад.

### "Пустой слайд"

"Пустой слайд" изображен на втором слайде [здесь](http://slides.com/pafik13/deck-1). По сути это темный фон с полупрозрачной кнопкой по центру, которая дает возможность добавить слайд. Такой слайд должен быть в конце каждой стопки слайдов внутри подгруппы.

### Верхняя боковая панель

В этой панели должно быть 4 кнопки:
1. Показать/скрыть панели - отвечает за отображение верхней и нижней панелей, причем как и в режиме редактирования, так и в режиме показа. То есть, если панели скрыты, то презентация представляет собой набор подряд идущих слайдов.
2. Добавить группу - добавляет группу после активной кнопки в верхней панели или в конец панели, если нет активных. После нажатия необходимо у пользователя запросить название группы и добавить её.
3. Редактировать группу - позволяет поменять название группы у активной кнопки в верхней панели. Данная клавиша активна только при активной/выбранной группе.
4. Удалить группу - позволяет удалить активную группы в верхней панели. Данная клавиша активна только при активной/выбранной группе.

### Нижняя боковая панель

В этой панели должно быть 3 кнопки:
1. Добавить подгруппу - добавляет подгруппу после активной кнопки в нижней панели или в конец панели, если нет активных. После нажатия необходимо у пользователя запросить название группы и добавить её. Активна только при активной/выбранной группе.
2. Редактировать подгруппу - позволяет поменять название подгруппы у активной кнопки в нижней панели. Данная клавиша активна только при активной/выбранной подгруппе.
3. Удалить подгруппу - позволяет удалить активную подгруппы в нижней панели. Данная клавиша активна только при активной/выбранной подгруппе.

**Comment by dashadasha172** [comment]
Вопрос, у каждой презентации свои группы? я просто думала группы можно в разные презентации добавлять, сейчас у них в БД стоит связь ManyToMany

**Comment by pafik13** [comment]
Хороший вопрос. Я бы не отрицал возможности повторного использования. Структуру лучше не менять, но на интерфейсе пока не надо придумывать возможность использования этого.
